src/bio/bio_buffer.c

```diff
@@ -1481,6 +1481,13 @@ bio_rwv(struct bio_io_context *ioctxt, struct bio_sglist *bsgl_in,
 	struct bio_desc		*biod;
 	int			 i, rc;
 
+	for (i = 0; i < bsgl_in->bs_nr; i++) {
+		if (bio_addr_is_hole(&bsgl_in->bs_iovs[i].bi_addr)) {
+			D_ERROR("Read/write a hole isn't allowed\n");
```

Comment:
I am some confused. If the object data contains hole, how can the application to read it? Before reading, how the application can know there is hole or not?

Reply:
bio_read/write() isn't exported to applications, they are called by DAOS server internally. The caller will pass in 'bsgl_in', so the caller must be able to know if there is any hole in the provided bsgl_in.